static/docs/commands-reference/commit.md

```diff
@@ -2,40 +2,212 @@
 
 Record changes to the repository by saving outputs to cache.
 
+## Synopsis
+
 ```usage
     usage: dvc commit [-h] [-q | -v] [-f] [-d] [-R] [targets [targets ...]]
 
     positional arguments:
       targets               DVC files.
-
-    optional arguments:
-      -f, --force           Commit even if checksums for dependencies/outputs
-                            changed.
-      -d, --with-deps       Commit all dependencies of the specified target.
-      -R, --recursive       Commit cache for subdirectories of the specified
-                            directory.
 ```
 
+## Description
+
+Normally DVC commands like `dvc add`, `dvc repro` or `dvc run`, commit the data
+to the DVC cache as the last step. What _commit_ means is that DVC
+
+* Computes a checksum for the file
+* Enters the checksum and file name into the DVC stage file
+* Tells the SCM to ignore the file (e.g. add entry to `.gitignore`).  If the
+  workspace was initialized with no SCM support (`dvc init --no-scm`) this does
+  not happen.
+* Adds the file to the DVC cache
+
+There are many cases where the last step is not desirable. For the DVC commands
+where it is appropriate the `--no-commit` option prevents the last step from
+occurring. The checksum is still computed and added to the DVC file, but the
+file is not added to the cache.
+
+That's where the `dvc commit` command comes into play. It handles that last
+step of adding the file to the DVC cache.
+
+The `dvc commit` command is useful for several scenarios where a stage is in
```

Comment:
I would mention that it can be used with `dvc add`-ed datasets/files/models to update them as a better alternative to run `dvc add` again.

Reply:
Okay